python/pyproject.toml

```diff
@@ -55,17 +49,11 @@ authors = [
 license = { text = "Apache 2.0" }
 requires-python = ">=3.9"
 dependencies = [
-    "cudf==24.6.*",
-    "cupy-cuda11x>=12.0.0",
-    "dask-cuda==24.6.*",
-    "dask-cudf==24.6.*",
+    "dask-cuda==24.6.*,>=0.0.0a0",
```

Comment:
I've forgotten how this is supposed to work -- didn't we need to preserve the suffixless packages like `cudf==24.6.*` in this file's default contents for the purpose of devcontainers?

Reply:
@trxcllnt can you attest to this?

Reply:
devcontainers don't care about the contents of pyproject.toml at all. What they care about is the contents of dependencies.yaml, specifically the requirements outputs (for pip devcontainers; conda devcontainers use the conda outputs). So what we need to maintain for devcontainers is a fallback for matrix entries of suffixed packages where the fallback is not suffixed.

That being said, I'm not sure I understand the rationale for the current contents of pyproject.toml. Why are so many dependencies removed from the default committed version of these lists? I feel like these lists should either contain all unsuffixed versions of every dependency to begin with, or these lists should be entirely empty and be filled in by RBB. Why this half-measure?

Reply:
The other relevant piece to be aware of for devcontainers is that in devcontainers we may need to control whether a package itself has a suffixed name or not. That's what https://github.com/rapidsai/rapids-build-backend/pull/17#discussion_r1584000078 is about. @trxcllnt can comment more on that use case.

Reply:
That's fair. I had originally considered that it might be more comprehensible to anyone actually inspecting the pyproject.toml file to see real-ish dependency lists, but maybe that doesn't matter.

Reply:
I concur with @vyasr’s earlier comment here. Having this list be “real-ish” is actually very helpful when explaining things to users/devs, even if it’s never used directly. I don’t like the awkward halfway-correct state.

Reply:
OK, I've added back the fake dependencies.

Reply:
Just adding the summary from our offline conversations last week, for those finding this thread in the future.

It's important (for RAPIDS devcontainers and DLFW builds) that the fallback matrices in `dependencies.yaml` have non-CUDA-specific dependencies. e.g. like this:

```yaml
- output_types: [requirements, pyproject]
  matrices:
    - matrix: {cuda: "12.*"}
      packages:
        - pylibraft-cu12==24.6.*,>=0.0.0a0
        - rmm-cu12==24.6.*,>=0.0.0a0
    - matrix: {cuda: "11.*"}
      packages:
        - pylibraft-cu11==24.6.*,>=0.0.0a0
        - rmm-cu11==24.6.*,>=0.0.0a0
    - matrix:
      packages:
        - pylibraft==24.6.*,>=0.0.0a0
        - rmm==24.6.*,>=0.0.0a0
```

As a side effect of how the `rapids-dependency-file-generator` pre-commit hook works across RAPIDS repos, that being true would also mean non-CUDA-specific dependencies in `pyproject.toml` in source control.